docs/grammar/SolidityParser.g4

```diff
@@ -272,8 +272,9 @@ locals [boolean constantnessSet = false, boolean visibilitySet = false, boolean
 		| {!$constantnessSet}? Constant {$constantnessSet = true;}
 		| {!$overrideSpecifierSet}? overrideSpecifier {$overrideSpecifierSet = true;}
 		| {!$constantnessSet}? Immutable {$constantnessSet = true;}
+		| {!$transientLocationSet}? Transient {$transientLocationSet = true;}
 	)*
-	name=identifier
+	(transientName=Transient | name=identifier)
```

Comment:
I was thinking that it would be needed in the case we had `transient` both as a data location and a variable name. But I guess I overlooked the fact that the flag `transientLocationSet` already takes care of not consuming it more than once as a data location.